pkg/spoke/controllers/finalizercontroller/manifestwork_finalize_controller.go

```diff
@@ -68,14 +68,17 @@ func (m *ManifestWorkFinalizeController) sync(ctx context.Context, controllerCon
 	// Delete appliedmanifestwork if relating manfiestwork is not found or being deleted
 	switch {
 	case errors.IsNotFound(err):
-		err := m.deleteAppliedManifestWork(ctx, appliedManifestWorkName)
+		err := m.deleteAppliedManifestWork(ctx, appliedManifestWorkName, false)
 		if err != nil {
 			return err
 		}
 	case err != nil:
 		return err
 	case !manifestWork.DeletionTimestamp.IsZero():
-		err := m.deleteAppliedManifestWork(ctx, appliedManifestWorkName)
+		// delete the applied manifestwork, but orphan the dependent resources, we will
+		// check the executor subject permission before deleting these resources
+		orphanDependents := manifestWork.Spec.Executor != nil
```

Comment:
hrm If we orphan delete the appliedwork, will lose the ownership of the resources, right? it could be conflict with deleteOption.